src/IK/include/BipedalLocomotion/IK/QPInverseKinematics.h

```diff
@@ -0,0 +1,130 @@
+ * inequality constraints.
+ * A possible usage of the IK can be found in "Romualdi et al. A Benchmarking of DCM Based
+ * Architectures for Position and Velocity Controlled Walking of Humanoid Robots"
+ * https://doi.org/10.1109/HUMANOIDS.2018.8625025
+ */
+class QPInverseKinematics : public InverseKinematics
+{
+    /**
+     * Private implementation
+     */
+    struct Impl;
+    std::unique_ptr<Impl> m_pimpl;
+
+public:
+
+    /**
+     * Constructor.
+     */
+    QPInverseKinematics();
+
+    /**
+     * Destructor.
+     */
+    ~QPInverseKinematics();
+
+    /**
+     * Add a task in the inverse kinematics
+     * @param task pointer to a given task
+     * @param priority is the priority of the task. The lower the number the higher the priority.
+     * @param weight weight associated to the task.
+     * @param isInequality true if the task should be treated as inequality constraint
+     * @note currently we support only task with priority 0 or 1. If the priority is set to 0 the
+     * task will be considered as equality constraint. This the weight is not required.
+     * @return true if the task has been added to the inverse kinematics.
+     */
+    bool addTask(std::shared_ptr<Task> task,
+                 const std::string& taskName,
+                 std::size_t priority,
+                 std::optional<Eigen::Ref<const Eigen::VectorXd>> weight = {},
+                 bool isInequality = false) override;
+
+    /**
+     * Finalize the IK.
+     * @param handler parameter handler.
+     * @note You should call this method after you add ALL the tasks.
+     * @return true in case of success, false otherwise.
+     */
+    bool finalize(const System::VariablesHandler& handler) override;
+
+
+    /**
+     * Initialize the planner.
+     * @param paramHandler pointer to the parameters handler.
+     * @note the following parameters are required by the class
+     * |           Parameter Name           |   Type   |                                       Description                                      | Mandatory |
+     * |:----------------------------------:|:--------:|:--------------------------------------------------------------------------------------:|:---------:|
+     * |    `robot_velocity_variable_name`  | `string` |   Name of the variable contained in `VariablesHandler` describing the robot velocity   |    Yes    |
+     * |             `verbosity`            |  `bool`  |                   Verbosity of the solver. Default value `false`                       |    No     |
+     * @return True in case of success, false otherwise.
+     */
+    bool initialize(std::weak_ptr<ParametersHandler::IParametersHandler> handler) override;
+
+    /**
+     * Get a vector containing the name of the tasks.
+     * @return an std::vector containing all the names associated to the tasks
+     */
+    std::vector<std::string> getTasksName() const override;
+
+    /**
+     * Return if the content of get is valid.
```

Comment:
I guess that a ``true`` is missing.

Reply:
Fixed in 5896d41fc6da6f533fccdcdb22a82c4d852d68ce